crates/orderbook/src/ipfs.rs

```diff
@@ -1,23 +1,40 @@
 use {
-    anyhow::{Context, Result},
+    anyhow::{anyhow, Context, Result},
     reqwest::{Client, ClientBuilder, StatusCode},
+    serde::{Deserialize, Serialize},
     std::time::Duration,
     url::Url,
 };
 
 pub struct Ipfs {
     client: Client,
     base: Url,
-    query: Option<String>,
+    auth_token: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct PinByHashRequest {
+    pub hash_to_pin: String,
+    pub pinata_metadata: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct PinResponse {
+    pub id: String,
+    pub ipfs_hash: String,
+    pub status: String,
+    pub name: Option<String>,
```

Comment:
The name is entirely unused by this code change and could be left out. Could use the hex appdata hash as the name.